app/components/Taps/TapConfiguration/MySQL/index.js

```diff
@@ -61,9 +69,24 @@ export default class MySQL extends Component<Props, MySQLState> {
     port: { validation: {}, errorMessage: 'Required' },
     user: { validation: {}, errorMessage: 'Required' },
     password: { validation: {}, errorMessage: 'Required' },
-    database: { validation: {}, errorMessage: 'Required' }
+    database: { validation: {}, errorMessage: 'Required' },
+    showModal: false,
+    checked: false
   };
 
+  componentWillMount() {
+    const { usesLogBaseRepMethod = false } = this.props.knotsStore;
+    this.setState(
+      {
+        checked: usesLogBaseRepMethod,
+        currentLogBasedBool: usesLogBaseRepMethod
+      },
+      () => {
+        this.props.updateLogBaseRepMethod(this.state.checked);
```

Comment:
This looks very strange.
Immediately on mount, a prop results in a state mutation which results in a redux store mutation which results in prop change?